crates/core/tedge/src/cli/mqtt/subscribe.rs

```diff
@@ -90,19 +90,8 @@ fn subscribe(cmd: &MqttSubscribeCommand) -> Result<(), anyhow::Error> {
         options.set_transport(rumqttc::Transport::tls_with_config(tls_config.into()));
     }
 
-    match cmd.duration {
-        Some(Duration::ZERO) | None => {}
-        Some(duration) if duration.as_secs() < 5 => {
-            options.set_keep_alive(duration);
-        }
-        Some(_) => {
-            options.set_keep_alive(Duration::from_secs(5));
-        }
-    }
-
     let (client, mut connection) = Client::new(options, DEFAULT_QUEUE_CAPACITY);
-    let interrupted = super::disconnect_if_interrupted(client.clone());
-    let started = std::time::Instant::now();
+    let interrupted = super::disconnect_if_interrupted(client.clone(), cmd.duration);
     let mut n_packets = 0;
```

Comment:
`messages` would be more descriptive of what the variable is actually counting ;)

```suggestion
    let mut n_messages = 0;
```

Reply:
done in 30906508844a93cf741398048174360e1f1da43d